File: server/app/schemas/analytics.py

from typing import Optional, List, Dict, Any
from pydantic import BaseModel
from datetime import datetime

class NewsItem(BaseModel):
    topic: str
    payload: Dict[str, Any]
    fetched_at: datetime

class FacultyAnalytics(BaseModel):
    faculty_id: str
    faculty_name: str
    department: Optional[str] = None
    verified_skills_count: int
    attempts_count: int
    total_enrollments: int
    avg_accuracy: float
    active_plan_progress: float
    top_gap: Optional[str] = None
    strengths: Optional[str] = None
    weaknesses: Optional[str] = None
    recommendations: List[str] = []
    ai_suggestion: Optional[str] = None
    ai_source: Optional[str] = None

class DepartmentSummary(BaseModel):
    department: str
    faculty_count: int
    avg_accuracy: float
    total_attempts: int
    plan_adoption_rate: float
    verified_skills_rate: float
    total_enrollments: int

File: client/src/lib/api/analytics.ts
import http from './http'

export interface FacultyAnalytics {
    faculty_id: string;
    faculty_name: string;
    department?: string;
    verified_skills_count: number;
    attempts_count: number;
    avg_accuracy: number;
    active_plan_progress: number;
}

export interface DepartmentSummary {
    department: string;
    faculty_count: number;
    avg_accuracy: number;
    total_attempts: number;
    plan_adoption_rate: number;
    verified_skills_rate: number;
    total_enrollments: number;
}

export const analyticsApi = {
    getDepartmentSummary: async (): Promise<DepartmentSummary[]> => {
        const response = await http.get<DepartmentSummary[]>('/api/v1/analytics/department-summary');
        return response.data;
    },

    getFacultyAnalytics: async (facultyId: string): Promise<FacultyAnalytics> => {
        const response = await http.get<FacultyAnalytics>(`/api/v1/analytics/faculty/${facultyId}`);
        return response.data;
    },

    getMyAnalytics: async (): Promise<FacultyAnalytics> => {
        const response = await http.get<FacultyAnalytics>('/api/v1/analytics/me');
        return response.data;
    }
}
